packages/framer-motion/src/frameloop/render-step.ts

```diff
@@ -1,16 +1,39 @@
 import { Step, Process } from "./types"
 
+class Queue {
+    order: Process[] = []
+    scheduled: Set<Process> = new Set()
+
+    add(process: Process) {
+        if (!this.scheduled.has(process)) {
+            this.scheduled.add(process)
+            this.order.push(process)
+            return true
+        }
+    }
+
+    remove(process: Process) {
+        const index = this.order.indexOf(process)
+        if (index !== -1) {
+            this.order.splice(index, 1)
+            this.scheduled.delete(process)
+        }
```

Comment:
I wonder if we should do a `this.scheduled.has` check first so we only run the indexOf lookup when needed, but if we're removing the lookup entirely soon then it probably doesn't matter as much.

Reply:
I think also remove happens so rarely it's probably not worth the check, whereas schedule can happen thousands of times a frame